filter/compare.go

```diff
@@ -250,6 +255,7 @@ func CompareFloat64(op string, pattern float64) (Predicate, error) {
 var compareString = map[string]func(string, string) bool{
 	"=":  func(a, b string) bool { return a == b },
 	"!=": func(a, b string) bool { return a != b },
+	"=~": func(a, b string) bool { return false },
```

Comment:
hmm actually for string don't we want to return `a == b` ?

Reply:
This is actually unreachable due to https://github.com/brimsec/zq/blob/pattern-match-operator/filter/compare.go#L445-L448
I had to put that check outside CompareBstring because we need to do the glob-regexp conversion before unescaping bstrings (ugh).  This could probably stand to be refactored to be clearer, but that snowballs into the whole topic of merging filters with expressions which is on hold indefinitely.